docs: fix/improve the v11.0.0 release info in `CHANGELOG.md`

gkalpak opened the pull request:
This PR includes several fixes/improvements to the v11.0.0 releade info in `CHANGELOG.md`. See the individual commits for more details.

##
While this PR manually fixes/improves the v11.0.0 release info, many of the fixes/improvements could be incorporated to our changelog generation tooling. This is being tracked in #39698.

petebacondarwin commented on CHANGELOG.md:
```diff
@@ -30,76 +29,76 @@
-* **router:** create schematic for preserveQueryParams ([#38762](https://github.com/angular/angular/issues/38762)) ([93ee05d](https://github.com/angular/angular/commit/93ee05d))
-* **router:** remove preserveQueryParams symbol ([#38762](https://github.com/angular/angular/issues/38762)) ([783a5bd](https://github.com/angular/angular/commit/783a5bd))
+* **core:** use Trusted Types policy in `createNamedArrayType()` ([#39209](https://github.com/angular/angular/issues/39209)) ([f6d5cdf](https://github.com/angular/angular/commit/f6d5cdf))
+* **router:** Allow undefined inputs on `routerLink` ([#39151](https://github.com/angular/angular/issues/39151)) ([b0b4953](https://github.com/angular/angular/commit/b0b4953))
```
```suggestion
* **router:** allow undefined inputs on `routerLink` ([#39151](https://github.com/angular/angular/issues/39151)) ([b0b4953](https://github.com/angular/angular/commit/b0b4953))
```

gkalpak replied:
😱

petebacondarwin commented on CHANGELOG.md:
```diff
@@ -121,19 +120,19 @@ because if `useAbsoluteUrl` is set to `true` then the behavior of the
+`RouterModule.forRoot` no longer supports `legacy_disabled`,
+`legacy_enabled`, `true`, or `false` as valid values.
+`legacy_enabled` (the old default) is instead `enabledNonBlocking`.
+* **router:** `enabled` is deprecated as a valid value for the
```
What happened here? Why did the generator miss the `**router:**`?

gkalpak replied:
The commit actually contains two breaking changes (as two bullet-points): https://github.com/angular/angular/commit/c4becca0e4238640461c43567a6d3e16b8c5d3f3

This showed up funny in the changelog, so I thought I''d list them as separate entries (mapped to the same commit), since they are two distinct changes.

petebacondarwin commented on CHANGELOG.md:
```diff
@@ -180,6 +149,54 @@ map both `null` and `undefined` to `null` and raise an exception on
+operator where necessary. In an unlikely case your code was testing the parent against `undefined` with strict equality,
+you'll need to change this to `=== null` instead, since the parent is now explicitly initialized
+with `null` instead of being left `undefined`.
+* **packaging:** In v10, IE 9, 10, and IE mobile support was deprecated. In v11, Angular framework removes IE 9,
```
Perhaps this should be called "browser-support", or perhaps "core", rather than "packaging"?

petebacondarwin reviewed (approved):
A few nits. Thanks for the tidy up!

petebacondarwin reviewed (approved):
Reviewed-for: public-api

gkalpak committed:
docs: fix typos in the v11.0.0 release info in `CHANGELOG.md`

This commit includes several minor fixes in the v11.0.0 release info in
`CHANGELOG.md`, such as:
- Fixing typos/punctuation.
- Making capitalization consistent.
- Wrapping code in backticks (for better readability).

gkalpak committed:
fixup! docs: fix typos in the v11.0.0 release info in `CHANGELOG.md`

gkalpak committed:
docs: remove commit with internal-only changes from `CHANGELOG.md`

Commit 4ca1c736bb020de166b6a1fbfb9fe264602ccd5c (part of PR #38931) was
incorrectly marked with scope `packaging`, while it is an internal-only
change (i.e. it does not affect packaging). As such is should have been
marked with a different scope (for example, `ci`/`build`/`refactoring`)
and should not appear in `CHANGELOG.md`.

This commit removes the entry from `CHANGELOG.md` to avoid confusing
developers trying to understand how this change may affect their apps.

NOTE:
The commit contains a breaking change notice (about dropping support for
IE<11) which _is_ relevant to app developers. This breaking change
notice is preserved in the "BREAKING CHANGES" section, but the scope is
changes from `packaging` to `core` to avoid confusion.

gkalpak committed:
fixup! docs: remove commit with internal-only changes from `CHANGELOG.md`

gkalpak committed:
docs: fix link to commit in `CHANGELOG.md`

Currently, the script that is used to generate the `CHANGELOG.md`
content uses the first 7 characters of a commit SHA in the generated
links. This is problematic when there are multiple commits in the repo
(or forks of the repo?) that start with the same 7 characters (which is
rare but possible), since GitHub shows a 404 page.

This was the case with commit 736e0644b02bc4606a7ae0c974d1b06e993708f6
that is included in the v11.0.0 release notes and happens to start with
the same characters as commit 736e064ac80f5e0ed84711694c2ba68809222ffd.

This commit fixes the issue by using the full commit SHA in the link for
this particular commit.

NOTE:
The underlying issue should be more generically addressed in the
changelog generation tooling. This is being tracked in #39698.

gkalpak committed:
docs: fix links to issues "closed" by a commit in the v11.0.0 release info in `CHANGELOG.md`

It seems that the script used to generate the `CHANGELOG.md` content
incorrectly treats any link or issue/PR reference in the commit message
as a link to an issue fixed/closed by the commit. However, there are
cases where such links/references do not refer to issues fixed/closed by
the commit (or not refer to issues at all).

This commit fixes such bogus links in the v11.0.0 release info.

NOTE:
The underlying issue should be more generically addressed in the
changelog generation tooling. This is being tracked in #39698.

gkalpak committed:
docs: group entries by scope in the v11.0.0 release info in `CHANGELOG.md`

Usually, entries in the release notes for a version in `CHANGELOG.md` are
grouped by scope (i.e. all `compiler` changes together, all `router`
changes together, etc.). Currently, the notes for a non-patch release
are created manually, by contatenating the release notes for all
corresponding next/RC versions. As a result, the entries for v11.0.0 are
not grouped together.

This commit fixes it by grouping all entries for the v11.0.0 release by
scope, making it easier to identify all changes that affect a specific
package/scope.

gkalpak committed:
docs: add links to corresponding commits in v11.0.0 breaking changes in `CHANGELOG.md`

This commit makes the breaking changes section of the v11.0.0 release
info in `CHANGELOG.md` easier to follow by:
1. Adding links to the corresponding commit for each breaking change (to
   make it easier to find the full context of the change).
2. Turning breaking changes of each scope into a bulleted list (to make
   it clearer that they affect the same package/scope).

NOTE:
It would be better if the changelog generation tooling handled this
automatically. This potential improvement is being tracked in #39698.

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

atscott reviewed (approved):
Reviewed-for: public-api
